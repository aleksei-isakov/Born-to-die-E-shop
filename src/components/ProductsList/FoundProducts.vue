<template>
  <p class="found-products">{{ productsQuantityValue }}</p>
</template>

<script>
export default {
  name: 'FoundProducts',

  props: {
    productsQuantity: {
      type: Number,
      required: true,
      default: 0
    }
  },

  computed: {
    productsQuantityValue() {
      return `Found: ${this.productsQuantity}`;
    }
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.found-products {
  font-weight: bold;
  color: $primary;
  margin: 20px 0;
  text-align: left;
}
</style>
