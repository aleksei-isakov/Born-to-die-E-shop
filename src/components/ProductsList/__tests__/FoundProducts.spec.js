import { mount } from '@vue/test-utils';
import FoundProducts from '@/components/ProductsList/FoundProducts.vue';

const productsQuantity = 20;

describe('FoundProducts', () => {
  let wrapper = mount(FoundProducts, {
    propsData: {
      productsQuantity
    }
  });

  it('should show the user the number of found products', () => {
    expect(wrapper.find('.found-products').text()).toBe(
      `Found: ${productsQuantity}`
    );
  });

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
