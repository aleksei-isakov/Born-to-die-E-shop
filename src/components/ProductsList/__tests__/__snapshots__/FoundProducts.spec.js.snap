// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoundProducts renders correctly 1`] = `
<p
  class="found-products"
>
  Found: 20
</p>
`;

exports[`FoundProducts should match snapshot 1`] = `
<p
  class="found-products"
>
  Found: 20
</p>
`;
